core/vm/interpreter.go

```diff
@@ -133,6 +141,11 @@ func (in *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err er
 	)
 	contract.Input = input
 
+	// Reclaim the stack as an int pool when the execution stops
+	defer func() {
+		poolOfIntPools.put(stackIP)
+	}()
```

Comment:
You most definitely don't want to push the entire stack into the pool. Most of the stacks will be empty, or containing just a few items. By pushing it into the intPoolPool, you are taking precious resource space from genuine int pools which have many big.Int's cached. Instead of doing this, just pop off the stack items and push them into `in.IntPool`.

Reply:
I see your point (at least I think I do): the pool will be small and more ints would have to be allocated when the next pool would have been sufficient.

The thing is, pushing on top of the other stack will also cause one allocation or more, depending on the capacity. And that pool could be re-used for another stack.

Which brings me to the following question: what about a pool of "shallow" stacks and one of "deep" stacks ? This way, we get some information about the pool's usage. Let me know if you think it's overkill.